test/WebJobs.Extensions.Tests/Extensions/NotificationHubs/NotificationHubAsyncCollectorTests.cs

```diff
@@ -41,12 +43,111 @@ public class NotificationHubAsyncCollectorTests
+
+            // Act
+            await collector.AddAsync(notification);
+
+            // Assert
+            mockNhClientService.VerifyAll();
+        }
+
+        [Fact]
+        public async Task AddAsync_SendNotification_EnableTestSend_Results_Null()
+        {
+            var notification = GetTemplateNotification();
+            // Arrange
+            var mockNhClientService = new Mock<INotificationHubClientService>(MockBehavior.Strict);
+
+            mockNhClientService.Setup(x => x.SendNotificationAsync(notification, "foo||bar"))
+                    .Returns(Task.FromResult(new NotificationOutcome()));
+
+            TestTraceWriter trace = new TestTraceWriter(TraceLevel.Info);
+
+            IAsyncCollector<Notification> collector = new NotificationHubAsyncCollector(mockNhClientService.Object, "foo||bar", true, trace);
 
             // Act
             await collector.AddAsync(notification);
 
             // Assert
+            Assert.Equal(2, trace.Events.Count);
+            Assert.True(trace.Events[0].Message.Contains("[NotificationHubsDebugLog] TrackingId"));
+            Assert.True(trace.Events[1].Message.Contains("[NotificationHubsDebugLog] NotificationOutcome.Results is null or empty"));
+        }
+
+        [Fact]
+        public async Task AddAsync_SendNotification_EnableTestSend_Results_NotNull()
+        {
+            var notification = GetTemplateNotification();
+            RegistrationResult reg = new RegistrationResult
+            {
+                ApplicationPlatform = "Windows",
+                Outcome = "Successfully sent Push notification",
+                PnsHandle = "Some-GUID",
+                RegistrationId = "Another-GUID"
+            };
+
+            var registrationList = new List<RegistrationResult>();
+            registrationList.Add(reg);
+
+            NotificationOutcome outcome = new NotificationOutcome
+            {
+                Results = registrationList,
+            };
+
+            string registrationOutcome = $"[NotificationHubsDebugLog] ApplicationPlatform:{reg.ApplicationPlatform} RegistrationId:{reg.RegistrationId} Outcome:{reg.Outcome}";
+
+            // Arrange
+            var mockNhClientService = new Mock<INotificationHubClientService>(MockBehavior.Strict);
+
+            mockNhClientService.Setup(x => x.SendNotificationAsync(notification, "foo||bar"))
+                    .Returns(Task.FromResult(outcome));
+
+            TestTraceWriter trace = new TestTraceWriter(TraceLevel.Info);
+
+            IAsyncCollector<Notification> collector = new NotificationHubAsyncCollector(mockNhClientService.Object, "foo||bar", true, trace);
+
+            // Act
+            await collector.AddAsync(notification);
+
+            // Assert
+            Assert.Equal(2, trace.Events.Count);
+            Assert.True(trace.Events[0].Message.Contains("[NotificationHubsDebugLog] TrackingId"));
```

Comment:
Rather than doing `.Contains` on all of these, can you do `.Equals()`? I think you had some bad spacing above, for example, and matching the whole string here would help catch that.